File: Python/6-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1 or len(s) <= numRows:
            return s

        rows = [''] * numRows
        curr = 0
        down = False

        for i in s:
            rows[curr] += i  
            if curr == 0 or curr == numRows - 1:  
                down = not down
            curr += 1 if down else -1

        return ''.join(rows)

File: Javascript/6_zigzag-conversion.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function (s, numRows) {
    if (numRows === 1 || s.length <= numRows) return s;

    let rows = new Array(numRows).fill('');
    let curr = 0;
    let down = false;

    for (let i = 0; i < s.length; i++) {
        rows[curr] += s[i]; 
        if (curr === 0 || curr === numRows - 1) {
            down = !down;
        }
        curr += down ? 1 : -1;
    }
    
    return rows.join('');
};